src/compiler/checker.ts

```diff
@@ -4973,6 +5031,7 @@ namespace ts {
                 if (relation === assignableRelation) {
                     if (isTypeAny(source)) return Ternary.True;
                     if (source === numberType && target.flags & TypeFlags.Enum) return Ternary.True;
+                    if (source.flags & TypeFlags.Boolean && target.flags & TypeFlags.Boolean && (source.flags & TypeFlags.PredicateType || target.flags & TypeFlags.PredicateType)) return Ternary.True;

```

Comment:
One interesting question is whether `boolean` or a type predicate should be assignable to a value type like `true`. Intuitively, it should not be.